### PR TITLE
Migrate request to axios for http origin

### PR DESCRIPTION
**Pull request by jdrydn**
Given that [`request/request`](https://github.com/request/request/issues/3142) has been deprecated for some time, here's a short PR to migrate the HTTP origin to [`axios/axios`](https://github.com/axios/axios) instead!

### CODE REVIEW

**Comment by jdrydn** [comment]
All three PRs, #9, #10 & #11 all include a version bump to v2 & enforce Node v10.x as standard.

File: src/origins/http.js [hunk 1]
```diff
@@ -1,55 +1,74 @@
...
-    return req(source);
-  });
+      } catch (err) {
+        if (err.response) {
+          const { status, statusText } = err.response || { status: 500, statusText: 'Internal Server Error' };
+          throw errors({
+            err: new Error(`${status} ${statusText} response from ${source}`),
+            status,
+          });
+        } else {
```
Comment (g-wilson) [inline src/origins/http.js:69]: Unnecessary else statement.

Consider inverting the conditional and return early? Or just remove the else.

File: src/origins/http.js [hunk 2]
```diff
@@ -1,55 +1,74 @@
...
+          // Always force a stream response
+          responseType: 'stream',
+          // Only allow an OK status or a redirect status
+          validateStatus: s => s === 200 || (s > 300 && s < 400),
+        });
+
+        if (status === 200) {
+          debug(`HTTP GET: ${source}: ${status} ${statusText}`, headers);
+          return data;
         } else {
```
Comment (g-wilson) [inline src/origins/http.js:56]: Again we can remove the else and clean it up a little i think 😊

**Review by g-wilson — APPROVED** [review]
Couple of code style comments

**Pull request merged**